lib/Backend/amd64/LowererMDArch.cpp

```diff
@@ -2888,6 +2888,109 @@ bool LowererMDArch::GenerateFastAnd(IR::Instr * instrAnd)
     return true;
 }
 
+bool LowererMDArch::GenerateFastDiv(IR::Instr * instrDiv)
+{
+    Assert(instrDiv);
+    auto * divisor = instrDiv->GetSrc2(); // denominator
+
+    if (PHASE_OFF(Js::BitopsFastPathPhase, this->m_func) || !divisor->IsIntConstOpnd() ||
+        !(instrDiv->m_opcode == Js::OpCode::Div_I4 || instrDiv->m_opcode == Js::OpCode::Rem_I4)) //ToDo Optimize unsigned division
+    {
+        return false;
+    }
+
+    auto * divident = instrDiv->GetSrc1(); // nominator
+    auto * dst = instrDiv->GetDst();
+    auto constDivisor = divisor->AsIntConstOpnd()->AsInt32();
+    auto isNegDevisor = false;
+
+    if (constDivisor < 0)
```

Comment:
why do this, it looks like the algorithm can handle negative divisors?

Reply:
Yes! this can be removed and is no longer needed. The algorithm will handle -ve divisors.

Reply:
Seems like we need this. The algorithm can handle -ve dividents, however for -ve const divisors we still need to negate the results.